samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerPlacementActions.java

```diff
@@ -1011,4 +1026,13 @@ private void assertBadRequests(String processorId, String destinationHost, Conta
     // Request shall be deleted as soon as it is acted upon
     assertFalse(containerPlacementMetadataStore.readContainerPlacementRequestMessage(requestMessage.getUuid()).isPresent());
   }
+
+  private void cleanUpRequestAllocatorThread(ContainerPlacementRequestAllocator requestAllocator, Thread containerPlacementRequestAllocatorThread) {
+    requestAllocator.stop();
+    try {
+      containerPlacementRequestAllocatorThread.join();
+    } catch (InterruptedException ie) {
+      Thread.currentThread().interrupt();
+    }
+  }
```

Comment:
Can this be part of `AfterMethod` tearDown provided you extract the request allocator to class level? It still gets cleaned up for tests that forget to invoke this explicitly.

Reply:
Not every test instantiates a ContainerPlacementRequestAllocator only two tests do hence this cleanup.

Also two tests use this individually, notice its not used in setup